test/org/zfin/antibody/api/AntibodyDetailsControllerTest.java

```diff
@@ -0,0 +1,163 @@
+import java.util.Optional;
+import java.util.Set;
+
+import static org.junit.Assert.*;
+
+@SuppressWarnings("SpringJavaInjectionPointsAutowiringInspection")
+@RunWith(SpringJUnit4ClassRunner.class)
+@WebAppConfiguration("home")
+@ContextConfiguration(classes = ZfinConfiguration.class)
+public class AntibodyDetailsControllerTest extends AbstractDatabaseTest {
+
+    @Autowired
+    private AntibodyAntigenGeneService service;
+
+    @Test
+    public void getAntigenGenes() {
+        // zn-5
+        String zdbID = "ZDB-ATB-081002-19";
+
+        List<MarkerRelationshipPresentation> results = service.getAntigenGenes(zdbID);
+
+//        assertEquals(1, results.size());
+        assertTrue(results.size() > 0);
+
+//        JsonResultResponse<MarkerDBLink> links = controller.addAntigenForAntibody(zdbID, formData)
+    }
+
+
+    @Test
+    public void getAntigenGenesWithMultiplePubs() {
+        String zdbID = "ZDB-ATB-090616-1";
+
+        List<MarkerRelationshipPresentation> results = service.getAntigenGenes(zdbID);
+
+        assertTrue(results.size() > 0);
+        Set<String> attributions = results.get(0).getAttributionZdbIDs();
+        assertEquals(2, attributions.size());
+
+    }
+
+    @Test
+    public void addAntigenGene() {
+        // zn-5
+        String zdbID = "ZDB-ATB-081002-19";
+
+        int originalCount = service.getAntigenGenes(zdbID).size();
+        service.addAntigenGeneForAntibody(zdbID,"alcamb", Set.of("ZDB-PUB-111111-1"));
+        int newCount = service.getAntigenGenes(zdbID).size();
+        assertEquals(originalCount + 1, newCount);
+
+        Optional<MarkerRelationshipPresentation> results = service.getAntigenGenes(zdbID)
+                .stream()
+                .filter(gene -> gene.getAbbreviation().equals("alcamb"))
+                .findFirst();
+
+        assertTrue(results.isPresent());
+        MarkerRelationshipPresentation alcamb = results.get();
+
+        Set<String> attributions = alcamb.getAttributionZdbIDs();
+        assertEquals(1, attributions.size());
+        assertTrue(attributions.contains("ZDB-PUB-111111-1"));
+    }
+
+    @Test
+    public void addAntigenGeneWithMultipleAttributions() throws Exception {
+        // zn-5
+        String zdbID = "ZDB-ATB-081002-19";
+
+        int originalCount = service.getAntigenGenes(zdbID).size();
+
+        //make changes
+        service.addAntigenGeneForAntibody(zdbID,"alcamb", Set.of("ZDB-PUB-111111-1", "ZDB-PUB-111110-1"));
+
+        //retrieve results to confirm
+        int newCount = service.getAntigenGenes(zdbID).size();
+        assertEquals(originalCount + 1, newCount);
+
+        List<String> attributions = service.getPublicationAttributionsForAntibodyAntigenGeneRelationship(zdbID, "alcamb");
+        assertEquals(2, attributions.size());
+
+        assertTrue(attributions.contains("ZDB-PUB-111111-1"));
+        assertTrue(attributions.contains("ZDB-PUB-111110-1"));
+    }
+
+    @Test
+    public void getAntigenGeneRelationshipo() {
```

Comment:
typo with the last character of the method name?

Reply:
Good catch